spec/controllers/idv/confirmations_controller_spec.rb

```diff
@@ -24,39 +26,67 @@
       init_idv_session
     end
 
-    describe 'all questions answered correctly' do
-      it 'shows success' do
+    context 'all answers correct' do
+      it 'activates profile' do
         complete_idv_session(true)
 
         get :index
-
-        expect(response.status).to eq 200
-        expect(response.body).to include(t('idv.titles.complete'))
-
         profile.reload
 
         expect(profile).to be_active
         expect(profile).to be_verified
       end
+
+      it 'redirects to original SAML Authn request' do
+        saml_authn_request = sp1_authnrequest
+        subject.session[:saml_request_url] = saml_authn_request
+        complete_idv_session(true)
+
+        get :index
+
+        expect(response).to redirect_to saml_authn_request
+      end
+
+      it 'cleans up PII from session' do
+        complete_idv_session(true)
+
+        get :index
+
+        expect(subject.user_session[:idv][:applicant]).to eq nil
+      end
+
+      context 'original SAML request is missing' do
+        it 'redirects to IdP profile' do
+          subject.session[:saml_request_url] = nil
+          complete_idv_session(true)
+
+          get :index
+
+          expect(response).to redirect_to(profile_index_path)
+        end
+      end
     end
 
-    describe 'some answers incorrect' do
+    context 'some answers incorrect' do
       it 'shows error' do
         complete_idv_session(false)
 
         get :index
 
         expect(response.status).to eq 200
         expect(response.body).to include(t('idv.titles.hardfail'))
+      end
 
-        profile.reload
+      it 'does not save PII' do
+        complete_idv_session(false)
+
+        get :index
 
-        expect(profile).to_not be_active
-        expect(profile).to_not be_verified
+        expect(Profile.where(id: profile.id).count).to eq 0
```

Comment:
I'm was going to say you convinced me with your argument and i'd be OK with the except :)     Either way is fine.